src/s3fetchers.ts

```diff
@@ -116,10 +123,13 @@ async function fetchSingleShard(
     const shard: Shard = await parseBody<Shard>(data.Body as Readable);
     return shard;
   } catch (err) {
-    console.error(
-      `Failed to fetch ${blockHeight}/shard_${shardId}.json. Retrying immediately`,
-      err
-    );
-    return await fetchSingleShard(client, bucketName, blockHeight, shardId);
+    if(retryCount > 0) {
+      console.warn(
+        `Failed to fetch ${blockHeight}/shard_${shardId}.json. Retrying in 200ms`,
+        err
+      );
+    }
+    await sleep(200);
+    return await fetchSingleShard(client, bucketName, blockHeight, shardId, retryCount++);
```

Comment:
Let's use a `while` loop just like it is done in `fetchBlock` so we don't introduce unnecessary recursion, and don't confuse users with the `retryCount` now being part of the interface (move it to be a local variable instead)

Reply:
Sounds good to me, will make the change